src/constants/gravityObjects/GravityObjects.ts

```diff
@@ -0,0 +1,62 @@
+export class GravityObject {
+  readonly name:string;
+  constructor(name?:string) {
+    this.name = name;
+  }
+
+  toString(): string {
+    return this.name;
+  }
+}
+
+export class AutoGravity {
+  readonly gravityObject:GravityObject;
+  private weight:number;
+  constructor(gravityObject:GravityObject, weight?: number) {
+    this.weight = weight;
+    this.gravityObject = gravityObject;
+  }
+
+  toString():string {
+    if(this.weight) {
+      const avoidStr = this.weight <=-100 ? 'avoid' : this.weight;
```

Comment:
1. what if weight===0?
2. I would put this logic in a separate function and return:
`${name}${getWeight()}

Reply:
1. What about it really? do we want to skip it? Is cat_0 not a valid input?
2. You'd like to see a function called 'concatNameAndWEight'? 

We could move all this logic (entirely) into a "calculateName or something similar, but then toString() will just call this function., feels a bit off.

wdyt?

Reply:
1. if(this.weight), will be false for weight===0, that's why i asked, what would you like to happen when weight=0?
2. I expect the void/anything else logic to get more complicated so i'd put the weight logic elsewhere, like calcWeight.

Reply:
I'm not sure it's well-defined anywhere, but a weight is always a number, `getWeight()` will always return a number. And if the user typed a number, we should keep it as a number in the URL.

The `avoid` keyword is for the other way around - it's there to make the user's life easier. using `avoid` without the need to understand weights. And in that case, we keep the word `avoid` as is. This is a strictly _server-side alias_ that we're not suppose to mess with - we should not replace one for the other on the client-side.

In other words, whatever the user did (either called it with an int weight, or invoked the `avoid()` function), whatever appears in the URL is exactly that - either the weight or the word `avoid`. No need to process anything.
In terms of internal implementation that probably means that you should keep another attribute besides the weight integer, and when building the string look at it first, and if nothing is there use the weight (or nothing at all if weight is null too). 

TL;DR 
`expect(Gravity.auto(AutoGravity.object(GravityObjects.CAT).avoid()).toString()).toBe('g_auto:cat_avoid');`
`expect(Gravity.auto(AutoGravity.object(GravityObjects.CAT, 50)).toString()).toBe('g_auto:cat_50');`
`expect(Gravity.auto(AutoGravity.object(GravityObjects.CAT, -100)).toString()).toBe('g_auto:cat_-100');`

Reply:
Fixed and added to the tests.